src/assisted_test_infra/download_logs/download_logs.py

```diff
@@ -174,13 +183,22 @@ def download_logs(
                     host["infra_env_id"], host["id"], os.path.join(output_folder, "cluster_files")
                 )
 
+        event_files = []
         for infra_env_id in infra_env_ids:
             with SuppressAndLog(assisted_service_client.rest.ApiException, KeyboardInterrupt):
-                client.download_infraenv_events(infra_env_id, get_infraenv_events_path(infra_env_id, output_folder))
+                infraenv_events = get_infraenv_events_path(infra_env_id, output_folder)
+                client.download_infraenv_events(infra_env_id, infraenv_events)
+                event_files.append(infraenv_events)
 
         with SuppressAndLog(assisted_service_client.rest.ApiException, KeyboardInterrupt):
-            client.download_cluster_events(cluster["id"], get_cluster_events_path(cluster, output_folder))
-            shutil.copy2(os.path.join(os.path.dirname(os.path.realpath(__file__)), "events.html"), output_folder)
+            cluster_events = get_cluster_events_path(cluster, output_folder)
+            client.download_cluster_events(cluster["id"], cluster_events)
+            event_files.append(cluster_events)
+
+            template = jinja2.Environment(
+                loader=jinja2.FileSystemLoader(searchpath=os.path.dirname(__file__))
+            ).get_template("events.html.j2")
+            template.stream(events=merge_events(event_files)).dump(os.path.join(output_folder, "events.html"))
```

Comment:
Maybe move this logic to a separate function? This method is already way too long..

Reply:
Done!